governance/src/agent.spec.js

```diff
@@ -0,0 +1,417 @@
+const {
+  Finding, createTransactionEvent, TransactionEvent, ethers, FindingType, FindingSeverity,
+} = require('forta-agent');
+
+const { provideHandleTransaction, provideInitialize } = require('./agent');
+const { getAbi } = require('./utils');
+const { createMockEventLogs, getObjectsFromAbi } = require('./test-utils');
+
+const config = require('../bot-config.json');
+
+const MINIMUM_EVENT_LIST = [
+  'ProposalCreated',
+  'ProposalCanceled',
+  'ProposalExecuted',
+];
+
+/**
+ * TransactionEvent(type, network, transaction, traces, addresses, block, logs, contractAddress)
+ */
+function createTxEvent({ logs, addresses }) {
+  return new TransactionEvent(null, null, null, [], [], null, logs, addresses);
+}
+
+// check the configuration file to verify the values
+describe('check bot configuration file', () => {
+  it('procotolName key required', () => {
+    const { protocolName } = config;
+    expect(typeof (protocolName)).toBe('string');
+    expect(protocolName).not.toBe('');
+  });
+
+  it('protocolAbbreviation key required', () => {
+    const { protocolAbbreviation } = config;
+    expect(typeof (protocolAbbreviation)).toBe('string');
+    expect(protocolAbbreviation).not.toBe('');
+  });
+
+  it('developerAbbreviation key required', () => {
+    const { developerAbbreviation } = config;
+    expect(typeof (developerAbbreviation)).toBe('string');
+    expect(developerAbbreviation).not.toBe('');
+  });
+
+  it('contracts key required', () => {
+    const { contracts } = config;
+    expect(typeof (contracts)).toBe('object');
+    expect(contracts).not.toBe({});
+  });
+
+  it('contracts key values must be valid', () => {
+    const firstContractName = Object.keys(config.contracts)[0];
```

Comment:
On one hand, we don't need to be specific about checking the first contract from the configuration file, because there should be only one anyway.

On the other hand, if you are already extracting the keys from the configuration `contracts` Object, it would seem natural to turn this into a for loop (or a forEach).